Alert Builder updates for GA and Alert Wizard fixes (#3506)
getsentry/sentry-docs: src/docs/product/sentry-basics/guides/alert-notifications/issue-alerts.mdx

Manage your alert rule by giving it a descriptive name. A descriptive alert rule name specifies the team affected and the topic of the alert. For example, "Frontend Latency," "Backend Failure Rate," or "Billing Apdex."

> You can choose a team to associate with the alert. When you do so, members of the selected team can edit this alert. Only teams that you're a member of are available for you to choose.

Reviewer: ```suggestion
You can choose a team to associate with an alert so that members of the that team can edit this alert. Note that you can only make this association if you are a member of the team.
```
